index.js

```diff
@@ -1,49 +1,34 @@
 'use strict';
 module.exports = (iterable, mapper, opts) => new Promise((resolve, reject) => {
-	const input = Array.from(iterable);
-
-	if (input.length === 0) {
-		resolve([]);
-		return;
-	}
-
 	opts = Object.assign({
-		concurrency: Infinity
+		concurrency: 1
 	}, opts);
```

Comment:
Since you can no more know the length of iterable upfront, you cannot correct `Infinity` concurrency. So, you have an option with a default value not allowed above (concurrency === Infinity).
A default value of 1 seem acceptable, what do you think?